cyfronet-fid/marketplace
# [#2543] API for bundles

## Add bundles to Offering API

Extend the Offering API maintaining backward compatibility.

The bundled_offers are identified in API by "<service.slug>/<offer.iid>"
to match their API identification.

## Add actions on bundle-relevant service and offer changes

In particular:
- unbundle offers that were made non-public and send notification to the
  remaining bundle offer then
- notify provider if her offer was bundled in another provider's bundle

Don't enforce rubocop's Lint/AmbiguousBlockAssociation in spec, as such
a syntax is a preferred way to express
`expect {...}.not_to change {...}` in rspec.

# How to test

Make sure to see whether [the assumptions from the issue](https://github.com/cyfronet-fid/marketplace/issues/2543#issuecomment-1043075327) hold both when using API and when triggering the emails from UI (as some mails are to be sent when an offer is unbundled due to, for example, being deleted/drafted).

Closes: #2543.
